topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlClient.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.cruisecontrol;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.strimzi.api.kafka.model.topic.KafkaTopic;
+
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Cruise Control REST API client.
+ * <br/><br/>
+ * The server runs one task execution at a time, additional 
+ * requests are queued up to {@code max.active.user.tasks}.
+ */
+public interface CruiseControlClient extends AutoCloseable {
+    /**
+     * HTTP request timeout in seconds.
+     */
+    long HTTP_REQUEST_TIMEOUT_SEC = 60;
+
+    /**
+     * Send a POST request to {@code topic_configuration} endpoint.
+     * This can be used to request replication factor changes.
+     * 
+     * @param kafkaTopics List of Kafka topics.
+     * @return The user task id.
+     */
+    String topicConfiguration(List<KafkaTopic> kafkaTopics);
+
+    /**
+     * Send a GET request to {@code user_tasks} endpoint.
+     * This can be used to check the asynchronous task execution result.
+     * 
+     * @param userTaskIds List of user task ids.
+     * @return User tasks response.
+     */
+    UserTasksResponse userTasks(List<String> userTaskIds);
```

Comment:
This contract is weird. The choice of collection in particular.. It requires a `List` (ordered, dupes allowed), but returns something containing a `List`, but the contract doesn't provide any guarantees about whether/how their orders are correlated. Either:

* The contract should guarantee that the returned List has the same number and ordering as the parameter
* Or the method should return something which contains a `Map<String, UserTask>`, allowing the called to pick out things based on the taskId, in which case this parameter can be relaxed to be a `Collection`.

Reply:
Looking at CC code, I can say that ordering is preserved and duplicates are filtered out. 

https://github.com/linkedin/cruise-control/blob/2.5.137/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/UserTaskState.java#L87-L113

Does ordering really matter here? Or is it just a problem with the contract? If it's the second, I'm switched back to using a Set as input parameter, which was my original choice.